test/project_types/script/forms/ask_app_test.rb

```diff
@@ -0,0 +1,81 @@
+# frozen_string_literal: true
+
+require "project_types/script/test_helper"
+
+describe Script::Forms::AskApp do
+  include TestHelpers::FakeFS
+  describe ".ask" do
+    let(:context) { TestHelpers::FakeContext.new(root: Dir.mktmpdir) }
+    let(:acting_as_shopify_organization) { false }
+    let(:selected_api_key) { "default_api_key" }
+    let(:selected_secret) { "default_secret" }
+
+    subject do
+      result = nil
+      capture_io do
+        result = Script::Forms::AskApp.ask(
+          context,
+          {
+            apps: apps,
+            acting_as_shopify_organization: acting_as_shopify_organization,
+          },
+          {}
+        )
+      end
+      result
+    end
+
+    def new_app(title, api_key, secret, app_type = "custom")
+      {
+        "title" => title,
+        "apiKey" => api_key,
+        "apiSecretKeys" => [{ "secret" => secret }],
+        "appType" => app_type,
+      }
+    end
+
+    describe("when asking app connection") do
+      describe("when number of apps == 0") do
+        let(:apps) { [] }
+
+        it("raises NoExistingAppsError") do
+          assert_raises(Script::Errors::NoExistingAppsError) { subject }
+        end
+      end
+
+      describe("when there is 1 public app") do
+        let(:apps) { new_app("app1", "1", "1", "public") }
+
+        it("raises NoExistingAppsError") do
+          assert_raises(Script::Errors::NoExistingAppsError) { subject }
+        end
+      end
+
+      describe("when number of apps == 1") do
+        let(:app_title) { "app1" }
+        let(:apps) { [new_app("app1", selected_api_key, selected_secret)] }
+
+        it("selects the app by default") do
+          selected_app_msg = context.message("script.application.ensure_env.app", app_title)
+          context.expects(:puts).with(selected_app_msg)
+          context.stubs(:puts).with(Not(equals(selected_app_msg)))
```

Comment:
These were copied over from the old ensure env test.
https://github.com/Shopify/shopify-cli/blob/main/test/project_types/script/tasks/ensure_env_test.rb#L183-L186

I think this last line is saying: stub out the context.puts method with any parameters that are not equal to the string.

I played around with commenting this out and also swapping it to an `expects`, and it seems this never gets hit.

```
unsatisfied expectations:
- expected exactly once, invoked never: #<TestHelpers::FakeContext:0x1590>.puts(Not("Partner organization {{green:business1 (1)}}."))
```

Digging deeper, I think this line was required in the old ensure_env way of doing things since the separate asks were all done at once. I think it is safe to remove this line from the refactor.